server/src/main/java/org/opensearch/common/metadata/MetadataContentParser.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.metadata;
+
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+
+import java.io.IOException;
+
+/**
+ * Parser interface for Metadata. Holds methods to convert to/from file content streams to metadata object holder - {@link T}
+ * @param <T> The type of metadata to be parsed
+ */
+public interface MetadataContentParser<T> {
```

Comment:
Does it need to be specific to `Metadata`?

Reply:
I had similar thoughts. This is generic enough to be used with anything that deals with indexInput and indexOuput. We can rename the classes and move to `common.io` package